crates/apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -192,15 +204,16 @@ where
             let tx_header = tx.header();
             // If [`process_proposal`] rejected a Tx, emit an event here and
             // move on to next tx
-            if ResultCode::from_u32(processed_tx.result.code).unwrap()
-                != ResultCode::Ok
-            {
-                let mut tx_event = new_tx_event(&tx, height.0);
-                tx_event["code"] = processed_tx.result.code.to_string();
-                tx_event["info"] =
-                    format!("Tx rejected: {}", &processed_tx.result.info);
-                tx_event["gas_used"] = "0".into();
-                response.events.push(tx_event);
+            if result_code != ResultCode::Ok {
+                response.events.emit(
+                    new_tx_event(&tx, height.0)
```

Comment:
not a deal breaker, but since everything begins with "with", why not just 
```
new_tx_event(&tx, height.0)
    .with(ResultCode(result_code)
    .with(Info(format!("Tx rejected: {}", &processed_tx.result.info)
    .with(GasUsed(0.into()),
```